### PR TITLE
Cleanup player references

### PR DESCRIPTION
**Pull request by jkasten2**
# Description
## 1 Line Summary
Remove code related to v15 APIs and behavior, such as player references.

## Details
Motivation to removing this deadcode is to make upcoming changes easier to make. This PR also results in a noticeable size reduction in the SDK.

### Scope
This PR addresses deleting anything related to v15 APIs and behavior. Some minor refactoring was done to clean up some related code paths.

# Validation
## Tests
Test on Windows 11 23H2 with Chrome 119.
* addTag
* Subscribing for notifications
* Clicking a notification
* Confirmed deliveries
* Session count and time
* Slidedown
* Bell
* Custom subscribe link

Tested on macOS 13.6.1 Safari 17.1
* Subscribing for notifications
* Clicking a notification
* Slidedown
* Bell
* Custom subscribe link

Tested on iOS 16.7.2
* Subscribing
* Clicking a notification
* Slidedown
* Bell
* Custom subscribe link

### Checklist
   - [X] All the automated tests pass or I explained why that is not possible
   - [X] I have personally tested this on my machine or explained why that is not possible
   - [X] I have included test coverage for these changes or explained why they are not needed

**Programming Checklist**
Interfaces:
   - [X] Don't use default export
   - [X] New interfaces are in model files

Functions:
   - [X] Don't use default export
   - [X] All function signatures have return types
   - [X] Helpers should not access any data but rather be given the data to operate on.

Typescript:
   - [X] No Typescript warnings
   - [X] Avoid silencing null/undefined warnings with the exclamation point

Other:
   - [X] Iteration: refrain from using `elem of array` syntax. Prefer `forEach` or use `map`
   - [X] Avoid using global OneSignal accessor for `context` if possible. Instead, we can pass it to function/constructor so that we don't call `OneSignal.context`

## Screenshots
Sizes of a production build in this PR:
```
webpack 5.80.0 compiled with 3 warnings in 8942 ms

─ ./build/bundles/OneSignalSDK.page.js
  ✔ ./build/bundles/OneSignalSDK.page.js            848B < 1 kB gzip  

─ ./build/bundles/OneSignalSDK.page.es6.js
  ✔ ./build/bundles/OneSignalSDK.page.es6.js        62.7KB < 79 kB gzip  

─ ./build/bundles/OneSignalSDK.sw.js
  ✔ ./build/bundles/OneSignalSDK.sw.js              35.33KB < 41 kB gzip  

─ ./build/bundles/OneSignalSDK.page.styles.css
  ✔ ./build/bundles/OneSignalSDK.page.styles.css    8.13KB < 9 kB gzip  

  4 checks passed   
```
### Checklist
   - [X] I have included screenshots/recordings of the intended results or explained why they are not needed

---

## Related Tickets

---

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Website-SDK/1146)

### CODE REVIEW

File: src/shared/api/OneSignalApiBase.ts [hunk 1]
```diff
@@ -145,6 +136,23 @@ export class OneSignalApiBase {
...
+  // have a app_id in the UUID format in the request
+  private static requestHasAppId(
+    url: string,
+    body?: Record<string, unknown>,
+  ): boolean {
+    if (url.startsWith('apps/')) {
+      const parts = url.split('/');
+      return isValidUuid(parts[1]);
+    }
+
```
Comment (jkasten2) [inline src/shared/api/OneSignalApiBase.ts:138]: Add comment, that the only thing using body `app_id` is the welcome notification.

File: src/shared/api/OneSignalApiBase.ts [hunk 2]
```diff
@@ -145,6 +136,23 @@ export class OneSignalApiBase {
       );
     }
   }
+
+  // OneSignal's backend requires that all request have a
+  // have a app_id in the UUID format in the request
+  private static requestHasAppId(
```
Comment (jkasten2) [inline src/shared/api/OneSignalApiBase.ts:130]: Add a unit test for this new method

File: src/shared/models/Session.ts [hunk 3]
```diff
@@ -20,9 +18,8 @@ export enum SessionOrigin {
 }
 
 export interface Session {
-  sessionKey: string;
+  sessionKey: string; // indexDb keyPath, always ONESIGNAL_SESSION_KEY
```
Comment (rgomezp) [inline src/shared/models/Session.ts:20]: if it is always the same string, why not restrict the type to that string?

**Review by rgomezp — APPROVED** [review]
Just a nit and what we discussed. Feel free to merge

**Pull request merged**